aws/internal/service/batch/equivalency/container_properties.go

```diff
@@ -19,10 +19,26 @@ func (cp *containerProperties) Reduce() error {
 		return aws.StringValue(cp.Environment[i].Name) < aws.StringValue(cp.Environment[j].Name)
 	})
 
+	if len(cp.Environment) == 0 {
```

Comment:
this is minor, but can you comment why you have to `nil` this out? I'm assuming due to marshaling?

Reply:
I'll put comments on each of these, thanks